### PR TITLE
Add ignore class id in loss/metrics

### PR DESCRIPTION
**Pull request by innat**
Fix https://github.com/innat/medic-ai/issues/77

## Summary by CodeRabbit

* **New Features**
  * Added target_class_ids and ignore_class_ids across losses and metrics; masking now flows through Dice/IoU/Tversky/GeneralizedDice and combined Dice+CE losses and metrics.
  * Cross-entropy and binary CE support optional masking for masked mean loss.

* **Bug Fixes**
  * Improved validation and clearer errors for class ID inputs.
  * Safer handling of ignored/empty classes and NaN prevention via mask-aware aggregations.

* **Chores**
  * Standardized public constructor parameters and added validations for unsupported ignore configurations.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: medicai/losses/dice_ce.py [hunk 1]
```diff
@@ -6,15 +6,26 @@
...
+        return ce_loss
 
-def binary_cross_entropy(y_true, y_pred):
+
+def binary_cross_entropy(y_true, y_pred, smooth=1e-7):
     spatial_dims = list(range(1, len(y_pred.shape) - 1))
+    y_pred = ops.clip(y_pred, smooth, 1.0 - smooth)
     ce_loss = -(y_true * ops.log(y_pred) + (1.0 - y_true) * ops.log(1.0 - y_pred))
     ce_loss = ops.mean(ce_loss, axis=spatial_dims)
     return ce_loss
```
[the remark this answers is not in the document]
Comment (innat) [reply medicai/losses/dice_ce.py]: /gemini 

Right. This was intentional. This Binary* loss also used for multi-label. And it wasn't clear how to handle multi-label cases for ingore_class_ids, any thoughts?
